lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,23 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      require Pathname.new(DEVICES_JSON_DIR).join("#{device}.rb").realpath.to_s
+
+      raise InvalidConfigFile, "#{device} does not exist" unless QemuHCK::Devices.respond_to?(device)
+
+      QemuHCK::Devices.public_send(device, @logger)
+    rescue Errno::ENOENT
```

Comment:
Don't rescue the whole method. It hides an error caused by `Models::QemuHCKDevice.from_json_file("#{__dir__}/amd-iommu.json", logger)`, for example.